server/dogger/views.py

```diff
@@ -175,25 +388,148 @@ def delete(self, request, pk, format=None):
+		schedules = SchedulesModel.objects.filter(walker=pk).values( 'day_of_week', 'hour'	 )
+		
+		print(schedules)
+		try:
+			return schedules
+		except Schedules.DoesNotExist:
+			raise Http404
+	@method_decorator(csrf_exempt)
+	def get(self, request, pk, format=None):
+		user = self.get_object(pk)
+		
+		print(user)
+		serializer = ScheduleSerializer(user, many =True	)
+
+		return Response(serializer.data)
+	@method_decorator(csrf_exempt)
+	def put(self, request, pk, format=None):
+		user = self.get_object(pk)
+		serializer = ScheduleSerializer(user, data=request.data)
+		if serializer.is_valid():
+			serializer.save()
+			return Response(serializer.data)
+		return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+	@method_decorator(csrf_exempt)
+	def delete(self, request, pk, format=None):
+		user = self.get_object(pk)
+		user.delete()
+		return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+
+class SchedulesDetailsForWalkerAndDayView(APIView):
+	"""
+	Retrieve, update or delete a schedule instance.
+	"""
+	@method_decorator(csrf_exempt)
+	def dispatch(self, request, *args, **kwargs):
+		return super().dispatch(request, *args, **kwargs)
+
+
+	@method_decorator(csrf_exempt)
+	def get_object(self, pk, day):
+		schedules = SchedulesModel.objects.filter(walker=pk, day_of_week = day).values('id', 'day_of_week', 'hour')
+		
+		try:
+			return schedules
+		except Schedules.DoesNotExist:
+			raise Http404
+	@method_decorator(csrf_exempt)
+	def get(self, request, pk, day, format=None):
+		user = self.get_object(pk, day)
+		
+		serializer = ScheduleSerializer(user, many =True	)
+
+		return Response(serializer.data)
+	@method_decorator(csrf_exempt)
+	def put(self, request, pk, format=None):
+		user = self.get_object(pk)
+		serializer = ScheduleSerializer(user, data=request.data)
+		if serializer.is_valid():
+			serializer.save()
+			return Response(serializer.data)
+		return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+	@method_decorator(csrf_exempt)
+	def delete(self, request, pk, format=None):
+		user = self.get_object(pk)
+		user.delete()
+		return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+
+class ScheduledWalksView(APIView):
+	"""
+	List all scheduled walks, create a new scheduled walk.
+	"""	
 	def get(self, request, format=None):
 		users = ScheduledWalksModel.objects.all()
 		serializer = ScheduledWalkSerializer(users, many=True)
 		return Response(serializer.data)
 
 	def post(self, request, format=None):
-		serializer = ScheduledWalkSerializer(data=request.data)
-		if serializer.is_valid():
-			serializer.save()
-			return Response(serializer.data, status=status.HTTP_201_CREATED)
-		return  Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-	
+		
+		datas = json.loads(request.body)
+		
+		user = WalkersModel.objects.get(pk = datas['walker'])
+		print(datas)
+		dog = DogsModel.objects.get(name = datas['dog'], owner = datas['owner'])
+		print("perro")
+		dog.save(walker=user) 

```

Comment:
Esto truena. No puedo registrar paseos
La manera correcta sería:
```suggestion
                dog.walker = user
		dog.save() 

```